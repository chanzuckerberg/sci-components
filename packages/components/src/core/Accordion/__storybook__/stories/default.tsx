import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
