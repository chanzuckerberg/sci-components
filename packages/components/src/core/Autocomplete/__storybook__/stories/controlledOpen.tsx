import { Args } from "@storybook/types";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { useState } from "react";
import RawAutocomplete from "src/core/Autocomplete";
import { AUTOCOMPLETE_WRAPPER_STYLES } from "../constants";

export const ControlledOpenDemo = <T extends DefaultAutocompleteOption>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
  } = props;

  const [open, setOpen] = useState(true);

  return (
    <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
      <RawAutocomplete
        onClickAway={() => {
          setOpen(false);
        }}
        onClick={() => {
          if (open) {
            setOpen(false);
          } else {
            setOpen(true);
          }
        }}
        open={open}
        id="autocomplete-demo"
        disableCloseOnSelect={multiple}
        label={label}
        multiple={multiple}
        options={options}
        getOptionDisabled={(option: T) => {
          return option.name === "Type: feature request";
        }}
        {...props}
      />
    </div>
  );
};
