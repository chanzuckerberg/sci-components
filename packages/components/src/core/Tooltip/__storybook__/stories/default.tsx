import { Args } from "@storybook/react";
import Button from "src/core/Button";
import RawTooltip from "src/core/Tooltip";

export const Tooltip = (props: Args): JSX.Element => {
  const { title } = props;
  return (
    <div>
      Hover over the info icon to view the tooltip.
      <p>
        ArrowOffset changes the position of the tooltip arrow and can be any
        numeric value within [-120, 120]. Any value value over the width of the
        tooltip will remove the arrow from the tooltip.
      </p>
      <div
        style={{
          margin: "135px 300px",
        }}
      >
        <RawTooltip title={title} {...props}>
          <Button
            aria-label="tooltip test button"
            sdsStyle="icon"
            sdsSize="large"
            icon="ExclamationMarkCircle"
          />
        </RawTooltip>
      </div>
    </div>
  );
};
