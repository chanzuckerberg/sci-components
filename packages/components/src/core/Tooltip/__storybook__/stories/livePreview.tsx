import ButtonIcon from "src/core/ButtonIcon";
import { TOOLTIP_LIVE_PREVIEW_STYLES } from "../constants";
import RawTooltip from "src/core/Tooltip";
import Button from "src/core/Button";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const LivePreviewDemo = (): JSX.Element => {
  return (
    <div style={TOOLTIP_LIVE_PREVIEW_STYLES as React.CSSProperties}>
      <RawTooltip
        title="Label lorem"
        sdsStyle="dark"
        placement="top"
        arrow
        open
      >
        <ButtonIcon
          sdsType="secondary"
          sdsSize="large"
          icon="InfoSpeechBubble"
        />
      </RawTooltip>
      <RawTooltip
        title={EXTRA_SHORT_LOREM_IPSUM}
        sdsStyle="light"
        placement="top"
        arrow
        open
      >
        <Button sdsType="primary" sdsStyle="rounded">
          Label
        </Button>
      </RawTooltip>
    </div>
  );
};
