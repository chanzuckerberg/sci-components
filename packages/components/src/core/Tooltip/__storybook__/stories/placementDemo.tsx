import { TOOLTIP_PLACEMENT_STYLES } from "../constants";
import RawTooltip from "src/core/Tooltip";
import Button from "src/core/Button";

export const PlacementDemo = (): JSX.Element => {
  return (
    <div style={TOOLTIP_PLACEMENT_STYLES as React.CSSProperties}>
      <RawTooltip title="Text" placement="top-start" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          top-start
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="top" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          top
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="top-end" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          top-end
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="left-start" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          left-start
        </Button>
      </RawTooltip>
      <div />
      <RawTooltip title="Text" placement="right-start" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          right-start
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="left" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          left
        </Button>
      </RawTooltip>
      <div />
      <RawTooltip title="Text" placement="right" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          right
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="left-end" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          left-end
        </Button>
      </RawTooltip>
      <div />
      <RawTooltip title="Text" placement="right-end" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          right-end
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="bottom-start" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          bottom-start
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="bottom" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          bottom
        </Button>
      </RawTooltip>
      <RawTooltip title="Text" placement="bottom-end" arrow open>
        <Button sdsStyle="minimal" sdsType="secondary">
          bottom-end
        </Button>
      </RawTooltip>
    </div>
  );
};
