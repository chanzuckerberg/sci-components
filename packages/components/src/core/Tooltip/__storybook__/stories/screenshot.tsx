import {
  TOOLTIP_MID_LABEL,
  TOOLTIP_ARROW_OFFSET_OPTIONS,
  TOOLTIP_ARROW_OPTIONS,
  TOOLTIP_DISPLAY_CONTENTS,
  TOOLTIP_SDS_STYLES,
  TOOLTIP_SUBTITLE_OPTIONS,
  TOOLTIP_WIDTH_OPTIONS,
} from "../constants";
import RawTooltip from "src/core/Tooltip";
import Button from "src/core/Button";

export const ScreenshotTestDemo = (): JSX.Element => {
  // loop through all SDS_STYLES
  return (
    <>
      {TOOLTIP_SDS_STYLES.map((sdsStyle) => {
        return <TooltipStyle sdsStyle={sdsStyle} key={sdsStyle} />;
      })}
    </>
  );

  // loop through all ARROW_OPTIONS + create headers for SDS_STYLES
  function TooltipStyle({
    sdsStyle,
  }: {
    sdsStyle: (typeof TOOLTIP_SDS_STYLES)[number];
  }) {
    const LEVEL_STYLE: React.CSSProperties = {
      columnGap: "25px",
      display: "inline-grid",
      fontFamily: "sans-serif",
      marginLeft: "50px",
    };
    const LABEL_STYLE: React.CSSProperties = {
      fontSize: "2em",
      gridColumn: "1 / 3",
      marginBottom: 0,
    };
    return (
      <div style={LEVEL_STYLE}>
        <p style={LABEL_STYLE}>
          Style: <b>{sdsStyle}</b>
        </p>
        {TOOLTIP_ARROW_OPTIONS.map((arrow) => {
          return (
            <TooltipArrow
              sdsStyle={sdsStyle}
              arrow={arrow}
              key={String(arrow)}
            />
          );
        })}
      </div>
    );
  }

  // loop through all ARROW_OFFSET_OPTIONS + create headers for ARROW_OPTIONS
  function TooltipArrow({
    sdsStyle,
    arrow,
  }: {
    sdsStyle: (typeof TOOLTIP_SDS_STYLES)[number];
    arrow: (typeof TOOLTIP_ARROW_OPTIONS)[number];
  }) {
    const LABEL_STYLE: React.CSSProperties = {
      ...TOOLTIP_MID_LABEL,
      borderWidth: "5px",
      fontSize: "1.5em",
      margin: "20px 0 0 0",
    };
    return (
      <div style={TOOLTIP_DISPLAY_CONTENTS}>
        <p style={LABEL_STYLE}>
          Arrow: <b>{arrow ? "true" : "false"}</b>
        </p>
        {TOOLTIP_ARROW_OFFSET_OPTIONS.map((arrowOffset) => {
          return (
            // arrowOffset is a dependency of arrow, so should only render if arrow is true
            ((arrow === true && arrowOffset !== undefined) ||
              (arrow === false && arrowOffset === undefined)) && (
              <TooltipArrowOffset
                sdsStyle={sdsStyle}
                arrow={arrow}
                arrowOffset={arrowOffset}
                key={String(arrowOffset)}
              />
            )
          );
        })}
      </div>
    );
  }

  // loop through all SUBTITLE_OPTIONS + create headers for ARROW_OFFSET_OPTIONS
  function TooltipArrowOffset({
    sdsStyle,
    arrow,
    arrowOffset,
  }: {
    sdsStyle: (typeof TOOLTIP_SDS_STYLES)[number];
    arrow: (typeof TOOLTIP_ARROW_OPTIONS)[number];
    arrowOffset: (typeof TOOLTIP_ARROW_OFFSET_OPTIONS)[number];
  }) {
    const LABEL_STYLE: React.CSSProperties = {
      ...TOOLTIP_MID_LABEL,
      borderWidth: "2px",
      fontSize: "1.17em",
      margin: "20px 0",
    };
    return (
      <div style={TOOLTIP_DISPLAY_CONTENTS}>
        {/* arrowOffset is a dependency of arrow, so should only render if arrow is true */}
        {arrow === true && (
          <p style={LABEL_STYLE}>
            Arrow offset: <b>{arrowOffset}</b>
          </p>
        )}
        {TOOLTIP_SUBTITLE_OPTIONS.map((subtitle) => {
          return (
            // subtitle is only allowed for dark style
            (sdsStyle === "dark" ||
              (sdsStyle === "light" && subtitle === undefined)) && (
              <TooltipSubtitle
                sdsStyle={sdsStyle}
                arrow={arrow}
                arrowOffset={arrowOffset}
                subtitle={subtitle}
                key={String(subtitle)}
              />
            )
          );
        })}
      </div>
    );
  }

  // loop through all WIDTH_OPTIONS + create headers for SUBTITLE_OPTIONS, WIDTH_OPTIONS
  function TooltipSubtitle({
    sdsStyle,
    arrow,
    arrowOffset,
    subtitle,
  }: {
    sdsStyle: (typeof TOOLTIP_SDS_STYLES)[number];
    arrow: (typeof TOOLTIP_ARROW_OPTIONS)[number];
    arrowOffset: (typeof TOOLTIP_ARROW_OFFSET_OPTIONS)[number];
    subtitle: (typeof TOOLTIP_SUBTITLE_OPTIONS)[number];
  }) {
    const WIDTH_LEVEL: React.CSSProperties = {
      display: "contents",
    };
    const LABEL_STYLE: React.CSSProperties = {
      fontSize: "0.67em",
      margin: "10px 0",
    };
    const DEFAULT_WIDTH_STYLE: React.CSSProperties = {
      margin: "0 125px 100px",
    };
    const WIDE_WIDTH_STYLE: React.CSSProperties = {
      margin: "0 275px 100px",
    };
    return (
      // assign specific styles to light-style Tooltip only, since it's the only one that has width options and needs special style-accommodations
      <div style={sdsStyle === "light" ? WIDTH_LEVEL : undefined}>
        {/* Show "Subtitle" label for dark-style Tooltip only */}
        {sdsStyle === "dark" && (
          <p style={LABEL_STYLE}>
            Subtitle: <b>{subtitle ? "yes" : "no"}</b>
          </p>
        )}
        {TOOLTIP_WIDTH_OPTIONS.map((width) => {
          return (
            // exclude "wide" width option for dark-style Tooltip
            (sdsStyle === "light" ||
              (sdsStyle === "dark" && width === "default")) && (
              <div key={String(sdsStyle) + String(width)}>
                <>
                  {/* Show "Width" label for light-style Tooltip only */}
                  {sdsStyle === "light" && (
                    <p style={LABEL_STYLE}>
                      Width: <b>{width}</b>
                    </p>
                  )}
                  <RawTooltip
                    // assign special styles depending on width option
                    style={
                      width === "default"
                        ? DEFAULT_WIDTH_STYLE
                        : WIDE_WIDTH_STYLE
                    }
                    title="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ut et labore dolore"
                    sdsStyle={sdsStyle}
                    subtitle={subtitle}
                    arrow={arrow}
                    arrowOffset={arrowOffset}
                    width={width}
                    key={String(width)}
                    open
                  >
                    <Button
                      aria-label="tooltip test button"
                      sdsStyle="icon"
                      sdsSize="medium"
                      icon="ExclamationMarkCircle"
                      sdsType="tertiary"
                      tabIndex={0}
                    ></Button>
                  </RawTooltip>
                </>
              </div>
            )
          );
        })}
      </div>
    );
  }
};
