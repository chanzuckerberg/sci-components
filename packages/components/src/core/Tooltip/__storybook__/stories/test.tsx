import { Args } from "@storybook/react";
import RawTooltip from "src/core/Tooltip";

export const TestDemo = (props: Args): JSX.Element => {
  const { title, ...rest } = props;
  return (
    <RawTooltip title={title} {...rest} data-testid="tooltip">
      <div style={{ display: "inline-block" }}>
        I am a tooltip child element.
      </div>
    </RawTooltip>
  );
};
