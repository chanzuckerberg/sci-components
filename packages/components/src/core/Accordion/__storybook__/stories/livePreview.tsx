import { Args } from "@storybook/react";
import { Accordion } from "./default";

const livePreviewStyles = {
  display: "grid",
  gridColumnGap: "24px",
  gridRowGap: "50px",
  gridTemplateColumns: "repeat(4, 200px)",
};

export function LivePreviewDemo(props: Args): JSX.Element {
  return (
    <div style={livePreviewStyles as React.CSSProperties}>
      <Accordion
        id="accordion-1"
        togglePosition="right"
        useDivider
        {...props}
      />
      <Accordion
        id="accordion-2"
        togglePosition="right"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
      <Accordion id="accordion-3" togglePosition="left" useDivider {...props} />
      <Accordion
        id="accordion-4"
        togglePosition="left"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
    </div>
  );
}
