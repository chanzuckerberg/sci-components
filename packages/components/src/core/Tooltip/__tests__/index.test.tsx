import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "../__storybook__/index.stories";

// Returns a component that already contain all decorators from story level, meta level and global level.
const { Test } = composeStories(stories);

describe("<Tooltip />", () => {
  generateSnapshots(stories);

  it("renders tooltip component", async () => {
    render(<Test {...Test.args} title="I am a tooltip" />);
    const tooltipElement = screen.getByTestId("tooltip");
    expect(tooltipElement).not.toBeNull();
  });

  it("does not render when no title content provided", () => {
    render(<Test {...Test.args} title="" />);
    const tooltipElement = screen.queryByTestId("tooltip");
    expect(tooltipElement).toBeNull();
  });

  it("renders children even when it does not render a tooltip", () => {
    render(<Test {...Test.args} title="" />);
    const child = screen.getByText("I am a tooltip child element.");
    expect(child).not.toBeNull();
  });
});
