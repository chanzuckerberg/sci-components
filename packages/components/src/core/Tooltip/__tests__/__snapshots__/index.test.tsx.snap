// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> Default story renders snapshot 1`] = `
<div>
  Hover over the info icon to view the tooltip.
  <p>
    ArrowOffset changes the position of the tooltip arrow and can be any numeric value within [-120, 120]. Any value value over the width of the tooltip will remove the arrow from the tooltip.
  </p>
  <div
    style="margin: 135px 300px;"
  >
    <button
      aria-label="tooltip test button"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium  css-zkd9l9-MuiButtonBase-root-MuiIconButton-root"
      data-mui-internal-clone-element="true"
      icon="ExclamationMarkCircle"
      tabindex="0"
      type="button"
    >
      <div
        class="css-1ft6wgl"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1q6hio5-MuiSvgIcon-root"
          data-file-name="IconExclamationMarkCircleLarge"
          data-testid="IconExclamationMarkCircleLarge"
          fillcontrast="white"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        />
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
